py/htm/examples/hotgym.py

```diff
@@ -158,17 +158,20 @@ def main(parameters=default_parameters, argv=None, verbose=True):
     tm_info.addData( tm.getActiveCells().flatten() )
 
     # Predict what will happen, and then train the predictor based on what just happened.
-    pdf = predictor.infer( count, tm.getActiveCells() )
-    for n in (1, 5):
-      if pdf[n]:
-        predictions[n].append( np.argmax( pdf[n] ) * predictor_resolution )
-      else:
-        predictions[n].append(float('nan'))
-    predictor.learn( count, tm.getActiveCells(), int(consumption / predictor_resolution))
-
-    anomalyLikelihood = anomaly_history.anomalyProbability( consumption, tm.anomaly )
-    anomaly.append( tm.anomaly )
-    anomalyProb.append( anomalyLikelihood )
+    if count > 5: #skip the n(=to the furthest predictions step) step, as predictor must learn something first
+      pdf = predictor.infer( tm.getActiveCells() )
```

Comment:
@ctrl-z-9000-times I dislike the requirement (learn before infer) I introduced in Predictor. For supervised tasks (MNIST) the API looks good. For unsupervised (Hotgym) not so much. 
What to do about that? 
- return `nan` until a predictor can output data (has learned least once, happens after `max(Predictor.steps)` steps)
  - disadvantage: this might need extra handling for plots etc, where one would like to expect result being a number (actually an integer, so not a nan either.)
- so what was the behavior before, return `0.0`, this is not 100% correct either, as `0` is a valid prediction, while for those numbers in should represent a `N/A`, but I guess it's acceptable compromise.

Reply:
That's a good idea. Instead of returning a failure result which may require special handling, it should return a result full of `nan's` to indicate that there is no valid data.

I would not worry about the plotting software. Most plotting software libraries (including python's matplotlib) understand that `nan` means missing data.